### PR TITLE
Bug 1288530 - Add a "classifier" field to PerformanceAlert, referencing the user model

### PR DESCRIPTION
**Pull request by crosscent**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1718)

### CODE REVIEW

**Comment by wlach** [comment]
This looks great! Could you make sure that the correct classifications / attributions are applied in  `tests/webapp/api/test_performance_alerts_api.py` (test that alerts are attributed to the "sheriff" user that classified them) and `tests/perfalert/test_create_alerts.py` (test that the alerts are not attributed to anyone)? You should be able to do this by extending the existing tests.

File: ui/partials/perf/alertsctrl.html [hunk 1]
```diff
@@ -115,11 +115,15 @@
           &nbsp;(<span ng-class="{'alert-invalid': alert.isInvalid(), 'alert-untriaged': alert.isUntriaged()}">{{alert.getStatusText()}}</span><span ng-show="alert.related_summary_id">
             <!-- Reassigned or downstream *to* another alert -->
             <span ng-if="alert.related_summary_id !== alertSummary.id">
-              to <a href="#/alerts?id={{alert.related_summary_id}}" target="_blank">alert #{{alert.related_summary_id}}</a>
+              to <a href="#/alerts?id={{alert.related_summary_id}}" target="_blank"
+                    uib-tooltip="{{alert.classifier? 'Classified by ' + alert.classifier: 'Classified automatically'}}"
```
Comment (wlach) [inline ui/partials/perf/alertsctrl.html]: I'd put a space between `alert.classifier` and `:`

File: ui/partials/perf/alertsctrl.html [hunk 2]
```diff
@@ -115,11 +115,15 @@
...
+                  >alert #{{alert.related_summary_id}}</a>
             </span>
             <!-- Reassigned or downstream *from* the another alert -->
             <span ng-if="alert.related_summary_id === alertSummary.id">
-              from <a href="#/alerts?id={{alert.summary_id}}" target="_blank">alert #{{alert.summary_id}}</a>
+              from <a href="#/alerts?id={{alert.summary_id}}" target="_blank"
+                    uib-tooltip="{{alert.classifier? 'Classified by ' + alert.classifier: 'Classified automatically'}}"
```
Comment (wlach) [inline ui/partials/perf/alertsctrl.html]: Ditto.

**Comment by crosscent** [comment]
Do the tests I've just added suffice? I hope I've added them to the right places. :)

**Comment by wlach** [comment]
Yes, this looks great. Will look into merging this tomorrow.

File: treeherder/webapp/api/performance_serializers.py [hunk 3]
```diff
@@ -50,6 +50,17 @@ def __init__(self, *args, **kwargs):
...
+    """Custom field for representing PerformanceAlert classifier"""
+    def to_representation(self, obj):
+        if obj is not None:
+            return obj.email
+        return obj
+
+    def to_internal_value(self, data):
+        return data
+
+
```
Comment (wlach) [inline treeherder/webapp/api/performance_serializers.py]: I am pretty sure that you should be able to implement this more simply with a slug related field: http://www.django-rest-framework.org/api-guide/relations/#slugrelatedfield

Please give it a try!

File: treeherder/webapp/api/performance_data.py [hunk 4]
```diff
@@ -253,6 +253,11 @@ class AlertPagination(pagination.CursorPagination):
 
     pagination_class = AlertPagination
 
+    def update(self, request, *args, **kwargs):
+        if 'related_summary_id' in request.data:
```
Comment (wlach) [inline treeherder/webapp/api/performance_data.py]: Take this line out to unconditionally set the classifier field in the case of an update. :)

File: ui/partials/perf/alertsctrl.html [hunk 5]
```diff
@@ -115,11 +115,15 @@
           &nbsp;(<span ng-class="{'alert-invalid': alert.isInvalid(), 'alert-untriaged': alert.isUntriaged()}">{{alert.getStatusText()}}</span><span ng-show="alert.related_summary_id">
             <!-- Reassigned or downstream *to* another alert -->
             <span ng-if="alert.related_summary_id !== alertSummary.id">
-              to <a href="#/alerts?id={{alert.related_summary_id}}" target="_blank">alert #{{alert.related_summary_id}}</a>
+              to <a href="#/alerts?id={{alert.related_summary_id}}" target="_blank"
+                    uib-tooltip="{{alert.classifier ? 'Classified by ' + alert.classifier : 'Classified automatically'}}"
```
Comment (wlach) [inline ui/partials/perf/alertsctrl.html]: Move this tooltip (with the updated logic I mentioned in the bug to not say "untriaged" alerts are autoclassified) up to the span just below the "alert-title" td.

**Pull request merged**